instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/tool_call_wrapper.py

```diff
@@ -0,0 +1,190 @@
+        if index < len(params):
+            param_name = params[index].name
+        _record_function_call_argument(span, param_name, entry, include_values)
+    for key, value in function_kwargs.items():
+        _record_function_call_argument(span, key, value, include_values)
+
+
+def _record_function_call_result(otel_wrapper, wrapped_function, result):
+    """Records the details about a function result as span attributes."""
+    include_values = is_content_recording_enabled()
+    span = trace.get_current_span()
+    span.set_attribute("code.function.return.type", type(result).__name__)
+    if include_values:
+        span.set_attribute(
+            "code.function.return.value", _to_otel_value(result)
+        )
+
+
+def _wrap_sync_tool_function(
+    tool_function: ToolFunction,
+    otel_wrapper: OTelWrapper,
+    extra_span_attributes: Optional[dict[str, str]] = None,
+    **kwargs,
+):
+    @functools.wraps(tool_function)
+    def wrapped_function(*args, **kwargs):
+        span_name = _create_function_span_name(tool_function)
+        attributes = _create_function_span_attributes(
+            tool_function, args, kwargs, extra_span_attributes
+        )
+        with otel_wrapper.start_as_current_span(
+            span_name, attributes=attributes
+        ):
+            _record_function_call_arguments(
+                otel_wrapper, tool_function, args, kwargs
+            )
+            result = tool_function(*args, **kwargs)
+            _record_function_call_result(otel_wrapper, tool_function, result)
+            return result
+
+    return wrapped_function
+
+
+def _wrap_async_tool_function(
+    tool_function: ToolFunction,
+    otel_wrapper: OTelWrapper,
+    extra_span_attributes: Optional[dict[str, str]] = None,
+    **kwargs,
+):
+    @functools.wraps(tool_function)
+    async def wrapped_function(*args, **kwargs):
+        span_name = _create_function_span_name(tool_function)
+        attributes = _create_function_span_attributes(
+            tool_function, args, kwargs, extra_span_attributes
+        )
+        with otel_wrapper.start_as_current_span(
+            span_name, attributes=attributes
+        ):
+            _record_function_call_arguments(
+                otel_wrapper, tool_function, args, kwargs
+            )
+            result = await tool_function(*args, **kwargs)
+            _record_function_call_result(otel_wrapper, tool_function, result)
+            return result
+
+    return wrapped_function
+
+
+def _wrap_tool_function(
+    tool_function: ToolFunction, otel_wrapper: OTelWrapper, **kwargs
+):
+    if inspect.iscoroutinefunction(tool_function):
+        return _wrap_async_tool_function(tool_function, otel_wrapper, **kwargs)
+    return _wrap_sync_tool_function(tool_function, otel_wrapper, **kwargs)
+
+
+def wrapped(
+    tool_or_tools: Optional[
+        Union[ToolFunction, ToolOrDict, ToolListUnion, ToolListUnionDict]
+    ],
+    otel_wrapper: OTelWrapper,
+    **kwargs,
```

Comment:
Can these kwargs be removed? I see we pass it along to the sub functions but it gets shadowed in the decorator functions